Update logic to open Aptoide when wallet no installed

mikieribeiro opened the pull request:
**What does this PR do?**

Verifies  aptoide's version and returns the appStoreIntent accordingly

The Install Wallet intent is automatically redirected to aptoide if its version is equal or greater than 9908 (Version Code). If lower, the user is given the opportunity to open the wallet's store page in either aptoide or google play

**Database changed?**

   No

**Where should the reviewer start?**

- helpers/WalletUtils.java
- manager/WalletUtils.java
- DialogWalletInstall.java

**How should this be manually tested?**

Test both poa (notification) and billing (install dialog) redirections to wallets page with no wallet installed using aptoide versions: 
--> greather than 9908 (automatic install should be done through Aptoide when the notification/install buttin is clicked)
--> lesser than 9908 (user should be given both options when the notification/install button is clicked, aptoide and google play)
 

  Flow on how to test this or QA Tickets related to this use-case: [APPC-1250](https://aptoide.atlassian.net/browse/APPC-<1250>)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1250](https://aptoide.atlassian.net/browse/APPC-<1250>)

**Questions:**

  No


**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

mikieribeiro committed:
Verifying aptoide's version and returning the appStoreIntent accordingly

-The Install Wallet intent is automatically redirected to aptoide if its version is equal or greater than 9908 (Version Code). If lower, the user is given the opportunity to open the wallet's store page in either aptoide or google play

mikieribeiro committed:
Change in gradle.properties and reformated code

-changed wallet's dev packageName back to .dev
-reformated code

JoaoRaimundo commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/WalletUtils.java:
```diff
@@ -33,6 +38,33 @@ public static boolean hasWalletInstalled() {
+        versionCode = pInfo.versionCode;
+      }
+
+      Log.d("AptoideVersion", "Version Code is: " + versionCode);
```
Do we really need this logging lines?

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -111,7 +139,7 @@ private static Intent getNotificationIntent() {
     intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
 
-    if (hasAptoide) {
+    if (hasAptoide && WalletUtils.getAptoideVersion() >= MINIMUM_APTOIDE_VERSION) {
```
do we need the hasAptoide sicne we already check the version and if not installed it return 0?

JoaoRaimundo commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/wallet/DialogWalletInstall.java:
```diff
@@ -178,9 +180,9 @@ private void redirectToStore() {
-
     final Intent appStoreIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(action));
-    if (WalletUtils.hasAptoideInstalled()) {
+    if (WalletUtils.hasAptoideInstalled()
```
Do we need to check if wallet is installed when we already have a check for the version that returns a value lower than the minim Aptoide's version?

mikieribeiro committed:
Removed unnecessary logs and the hasAptoideInstalled method

-removed version code logs
-removed hasAptoideInstalled method because the getApoideVersion method already verifies aptoide's installation on the device

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -38,26 +43,39 @@ public static void setContext(Context cont) {
-      packageManager.getPackageInfo(BuildConfig.BDS_WALLET_PACKAGE_NAME, 0);
-      return true;
+      pInfo = context.getPackageManager()
+          .getPackageInfo(com.appcoins.sdk.android.billing.BuildConfig.APTOIDE_PACKAGE_NAME, 0);
```
Why do we need to access to android buildConfigs that are not for the module we are using?
What of the developer imports only the Ads SDK?

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -38,26 +43,39 @@ public static void setContext(Context cont) {
-      return true;
+      pInfo = context.getPackageManager()
+          .getPackageInfo(com.appcoins.sdk.android.billing.BuildConfig.APTOIDE_PACKAGE_NAME, 0);
+      String versionName = pInfo.versionName;
```
The version name is not being used, it can be removed.

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -93,25 +111,24 @@ public static void createInstallWalletNotification() {
 
-    if (hasAptoide) {
+
+    if (WalletUtils.getAptoideVersion() != UNINSTALLED_APTOIDE_VERSIONCODE) {
```
Do we need to get Aptoide version to check if it is installed? If so then the variable used as comparison (UNINSTALLED_APTOIDE_VERSIONCODE) then it should be set as the default value in the getAptoideVersion method.

mikieribeiro committed:
Removed unnecessary code and renamed unmatching and confusing variable names

-Removed Aptoide's versionName
-Renamed versionCode to
DEFAULT_APTOIDE_VERSIONCODE

mikieribeiro committed:
Variable refactoring and method integration

-Made the UNINSTALLED_APTOIDE_VERSION_CODE default on the getAptoideVersion() method so the value zero is returned when Aptoide is not installed

JoaoRaimundo committed:
[APPC-1250] Solve conflicts.
